packages/discord.js/src/structures/Webhook.js

```diff
@@ -7,6 +7,8 @@ const MessagePayload = require('./MessagePayload');
 const { Error } = require('../errors');
 const DataResolver = require('../util/DataResolver');
 
+let Message;
```

Comment:
Completely out of curiosity, can we not do something like 
```js
const Message = (() => require('./Message').Message)();
```
for lazy loading?

Reply:
Then can't we just do?
```js
const { Message } = require('./Message');
```

Reply:
If it's done like its currently done in the PR it's because there's circular imports at hand. So..no

Reply:
> Completely out of curiosity, can we not do something like
> 
> ```js
> const Message = (() => require('./Message').Message)();
> ```
> 
> for lazy loading?

Doesn't seem to work, still gives me a circular import warning.

Reply:
Circular dependencies are triggered when A requires B, B requires A, and A is empty.

This is why many files define their requires at the bottom, right after `module.exports =` is defined, so A isn't empty, B can resolve A, and then A can resolve B.

A prettier alternative is to make a `lazy` method utility (similar to Rust's `lazy_static`, sorta):

```typescript
export function lazy<T>(cb: () => T) {
	let defaultValue: T;

	return () => (defaultValue ??= cb());
}
```

And then we do the following:

```js
const getMessage = lazy(() => require('./Message').Message);
```

Then once we do `getMessage();`, it gets cached and has no performance impact over the next few operations. This approach also deduplicates requires and is cleaner:

```diff
- let Message;
+ const getMessage = lazy(() => require('./Message').Message);
  // ...
- Message ??= require('./Message').Message;
+ const Message = getMessage();
```

Reply:
Does it make any performance difference for node.js requires?

Node.js caches modules after the first time they're loaded anyway:

> Modules are cached after the first time they are loaded. This means (among other things) that every call to require('foo') will get exactly the same object returned, if it would resolve to the same file.

> Provided require.cache is not modified, multiple calls to require('foo') will not cause the module code to be executed multiple times.

Reply:
Yes, because it needs to resolve a relative path into an absolute one, among many other operations.

Reply:
Even with the `lazy()` function I still get a circular import error.

Reply:
Then you're using it wrong.

You're supposed to use it inside the code that needs it, not at the very top, which would be the same as having a top-level require.